jax/numpy/lax_numpy.py

```diff
@@ -1228,6 +1228,22 @@ def nan_reduction(a, axis=None, out=None, keepdims=False, **kwargs):
 nansum = _make_nan_reduction(onp.nansum, sum, 0, nan_if_all_nan=False)
 nanprod = _make_nan_reduction(onp.nanprod, prod, 1, nan_if_all_nan=False)
 
+@_wraps(onp.nanmean)
+def nanmean(a, axis=None, dtype=None, out=None, keepdims=False):
+  if out is not None:
+    raise ValueError("nanmean does not support the `out` argument.")
+  if (onp.issubdtype(lax._dtype(a), onp.bool_) or
+      onp.issubdtype(lax._dtype(a), onp.integer)):
+    return mean(a, axis, dtype, out, keepdims)
+  # Check and Count all non-NaN values
+  nan_mask = logical_not(isnan(a))
+  normalizer = sum(nan_mask, axis=axis, dtype=int32, keepdims=keepdims)
+  normalizer = lax.convert_element_type(normalizer, dtype)
+  if dtype is None:
+    dtype = lax._dtype(a)
+  #Perform mean calculation
+  td = true_divide(nansum(a, axis, dtype=dtype, keepdims=keepdims), normalizer)
+  return lax.convert_element_type(td, dtype)
```

Comment:
I believe this final `lax.convert_element_type` is redundant; the type of the divide output is always guaranteed to be `dtype`.

You can probably also write `lax.div` instead of `true_divide` here. They will  do the same thing given that both inputs are already of the correct `dtype`, and we may as well call the lower-level `lax` primitives where they suffice.